### DISCUSSION TITLE
How to use OpenAI Function Agent with Vector Store Retriever?

### DISCUSSION THREAD

**Discussion by room3dev** [FlowiseAI/Flowise#678] [Q&A] [answered]
I want to create Q & A Chatbot using data from Vector Store Retriever with ability to calling functions.
I can do it separately but have no Idea how to put them together. any help/idea will be appreciated. 🙏
![image](https://github.com/FlowiseAI/Flowise/assets/25867378/7cbe4532-ff34-4f1e-9703-d50fa7e6480f)

**Comment by FenixOrdern** [comment, marked as answer]
Use the ChainTool as a wrapper for your QA chain, connect it to the Function Agent on the Allowed Tools line

**Comment by HenryHengZJ** [comment]
We have a new conversational retrieval agent coming up in next release that allow you to use Vector Store Retriever as Tool to OpenAI Agent for function calling

**Reply by zhanghenryu** [reply]
hello , can the chain tools show the source document of QA Chain ?

 I got the data in the document and a "output values have 2 keys, you must specify an output key or pass only 1 key as output" warning .
